src/web/app/pages-instructor/instructor-course-enroll-page/instructor-course-enroll-page.component.ts

```diff
@@ -94,59 +98,89 @@ export class InstructorCourseEnrollPageComponent implements OnInit {
     this.enrollErrorMessage = '';
+    this.allStudentChunks = [];
+
     const newStudentsHOTInstance: Handsontable =
         this.hotRegisterer.getInstance(this.newStudentsHOT);
-
     const hotInstanceColHeaders: string[] = (newStudentsHOTInstance.getColHeader() as string[]);
 
-    const studentsEnrollRequest: StudentsEnrollRequest = {
-      studentEnrollRequests: [],
-    };
+    let currentStudentChunk: StudentEnrollRequest[] = [];
 
     // Parse the user input to be requests.
     // Handsontable contains null value initially,
     // see https://github.com/handsontable/handsontable/issues/3927
     newStudentsHOTInstance.getData()
         .filter((row: string[]) => (!row.every((cell: string) => cell === null || cell === '')))
-        .forEach((row: string[]) => (studentsEnrollRequest.studentEnrollRequests.push({
-          section: row[hotInstanceColHeaders.indexOf(this.colHeaders[0])] === null ?
-              '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[0])].trim(),
-          team: row[hotInstanceColHeaders.indexOf(this.colHeaders[1])] === null ?
-              '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[1])].trim(),
-          name: row[hotInstanceColHeaders.indexOf(this.colHeaders[2])] === null ?
-              '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[2])].trim(),
-          email: row[hotInstanceColHeaders.indexOf(this.colHeaders[3])] === null ?
-              '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[3])].trim(),
-          comments: row[hotInstanceColHeaders.indexOf(this.colHeaders[4])] === null ?
-              '' : row[hotInstanceColHeaders.indexOf(this.colHeaders[4])].trim(),
-        })));
-
-    this.studentService.enrollStudents(
-        this.courseId, studentsEnrollRequest,
-    ).pipe(finalize(() => this.isEnrolling = false)).subscribe((resp: Students) => {
-      const enrolledStudents: Student[] = resp.students;
-      this.showEnrollResults = true;
-      this.statusMessage.pop(); // removes any existing error status message
-      this.statusMessageService.showSuccessToast('Enrollment successful. Summary given below.');
-      this.enrollResultPanelList =
-          this.populateEnrollResultPanelList(this.existingStudents, enrolledStudents,
-              studentsEnrollRequest.studentEnrollRequests);
-    }, (resp: ErrorMessageOutput) => {
-      this.enrollErrorMessage = resp.error.message;
-    }, () => {
-      this.studentService.getStudentsFromCourse({ courseId: this.courseId }).subscribe((resp: Students) => {
-        this.existingStudents = resp.students;
-        if (!this.isExistingStudentsPanelCollapsed) {
-          const existingStudentTable: Handsontable = this.hotRegisterer.getInstance(this.existingStudentsHOT);
-          this.loadExistingStudentsData(existingStudentTable, this.existingStudents);
-        }
-        this.isExistingStudentsPresent = true;
-      });
+        .forEach((row: string[]) => {
+          currentStudentChunk.push({
```

Comment:
Possibly do a check here to see if the row in the table is empty first before adding it in?